src/components/node-list/node-list-items.js

```diff
@@ -159,13 +165,63 @@ export const getFilteredModularPipelineItems = createSelector(
         faded: false,
         visible: true,
         disabled: false,
-        unset: false,
         checked: true,
       })
     ),
   })
 );
 
+/**
+ * Return filtered/highlighted element types
+ * @param {string} searchValue Search term
+ * @return {object} Grouped element types
+ */
+export const getFilteredElementTypes = createSelector(
+  [(state) => state.searchValue],
+  (searchValue) =>
+    highlightMatch(
+      filterNodeGroups(
+        {
+          elementType: Object.entries(sidebarElementTypes).map(([type, name]) => ({
+            id: type,
+            name,
+          })),
+        },
+        searchValue
+      ),
+      searchValue
+    )
+);
+
+/**
+ * Return filtered/highlighted element type items
+ * @param {object} filteredTags List of filtered element types
+ * @param {array} nodeTypes List of node types
+ * @return {object} Element type items
+ */
+export const getFilteredElementTypeItems = createSelector(
+  [getFilteredElementTypes, (state) => state.nodeTypes],
+  (filteredElementTypes, nodeTypes) => ({
+    elementType: filteredElementTypes.elementType.map((elementType) => {
```

Comment:
I wonder what is the reasoning behind wrapping the returned results from `getFilteredElementTypes` under the `elementType` field in the first place, as the name of this function suggests it is only returning results based on element types?

Reply:
This is just to keep consistency with the same existing pattern used for all the other filters in this file (filter inputs -> create sidebar items). I agree though it would be nice to refactor this but at this point I think the priority is to merge this first?

Reply:
I see - yea I agreem we can refactor this away in a future PR.